### DISCUSSION TITLE
Rendering Video and Model Seperately

### DISCUSSION THREAD

**Discussion by Akhil-coditas** [google/model-viewer#4284] [Q&A]
How i can render a video after clicking ar button..i have video of gate opening with green screen..i want to display video after clicking the ar button and then when gate is opening the camera should be visible with the model insplace of green screen....i tried using video texture..but then the video where gate is opening is being displayed on the model. not as a separate video

**Comment by elalish** [comment]
Yes, that's what a video texture is. I think your questions are more about general web design than MV specifically. Try to get all the non-3D stuff working (put temporary images in place of the MV elements) and then come back with any questions specific to the 3D parts.

**Reply by Akhil-coditas** [reply]
so basically.. i need video and model to load seperately..after clicking on ar button. like video should play first (gate opening video) then while gate is opening the camera with model should be shown( i have video with gate opening with green screen where i need to place camera and model). is it possible with model-viewer?

**Reply by elalish** [reply]
Not sure exactly what you mean by green screen - if you're hoping to do fancy compositing between the video and the render, that may be pretty tricky. You might consider using a WebM video, so you can encode alpha transparency into the video and skip the green. I can't say much more without seeing examples.

**Reply by Akhil-coditas** [reply]
https://github.com/google/model-viewer/assets/106797543/8c00393e-cb17-4a28-8eac-a4e53d9e9211
requirment - 
on clicking activate button i need to show door video... and then my custom video .
in between door video where you see the green part , camera must be shown with the custom video in ar
and my door video should take the whole screen size so that it looks like door is opening and my custom video is being played inside it on ar . hope you got the requirement.

 i will use webm video later though as you suggested but for now 


what im doing is im loading door video first on my model...then im replacing door video with the custom video.

so how do i load door video which will take entire screen size..and then when the door opening video is completed..my custom video must be played in center in ar
